stdlib/javascript/compile.mc

```diff
@@ -121,56 +210,104 @@ lang MExprJSCompile = MExprAst + JSProgAst
 
       -- Intrinsics
       else match fun with TmConst { val = val } then
-        let args = map compileExpr args in
-        compileOp fun args val
+        let args = map (compileMExpr opts) args in
+        compileCOp opts args val
 
-      else error "Unsupported application in compileExpr"
+      else error "Unsupported application in compileMExpr"
     else never
 
   -- Anonymous function, not allowed.
-  | TmLam _ -> error "Anonymous function in compileExpr."
+  | TmLam { ident = arg, body = body } ->
+    let body = (compileMExpr opts) body in
+    -- dprintLn "Compiling lambda";
+    -- dprintLn body;
+    JSEFun { param = arg, body = body }
 
   -- Unit type is represented by int literal 0.
   | TmRecord { bindings = bindings } ->
-    if mapIsEmpty bindings then JSEInt { i = 0 }
-    else error "ERROR: Records cannot be handled in compileExpr."
+    let fieldSeq = mapToSeq bindings in
+    let compileField = lam f. match f with (sid, expr)
+      then (sidToString sid, (compileMExpr opts) expr)
+      else never in
+    JSEObject { fields = map compileField fieldSeq }
 
-  | TmSeq {tms = tms, ty = ty, info = info} & t ->
-    -- Special handling of strings
+  | TmSeq {tms = tms} ->
     -- Check if sequence of characters, then concatenate them into a string
     if _isCharSeq tms then
       match (_charSeq2String tms) with Some str then JSEString { s = str }
-      else infoErrorExit (infoTm t) "Non-literal strings currently unsupported."
+      else never
     else
-      -- infoErrorExit (infoTm t) "Non-literal strings currently unsupported."
-      -- Else compile each expression in sequence and return a list
-      let tms: [JSExpr] = map compileExpr tms in
-      JSESeq { exprs = tms, info = info }
-
-  | TmRecordUpdate _ -> error "Record updates cannot be handled in compileExpr."
-  | TmConApp _ -> error "Constructor application in compileExpr."
-  | TmLet _ -> error "Let expressions cannot be handled in compileExpr."
-  | TmRecLets _ -> error "Recursive let expressions cannot be handled in compileExpr."
-  | TmType _ -> error "Type expressions cannot be handled in compileExpr."
-  | TmConDef _ -> error "Constructor definitions cannot be handled in compileExpr."
-  | TmMatch _ -> error "Match expressions cannot be handled in compileExpr."
-  | TmUtest _ -> error "Unit test expressions cannot be handled in compileExpr."
-  | TmExt _ -> error "External expressions cannot be handled in compileExpr."
+      JSEArray { exprs = map (compileMExpr opts) tms }
 
   -- Literals
   | TmConst { val = val } ->
     match val      with CInt   { val = val } then JSEInt   { i = val }
     else match val with CFloat { val = val } then JSEFloat { f = val }
     else match val with CChar  { val = val } then JSEChar  { c = val }
     else match val with CBool  { val = val } then JSEBool  { b = val }
+    else match compileCOp opts [] val with jsexpr then jsexpr -- SeqOpAst Consts are handled by the compile operator semantics
```

Comment:
This `match` will always succeed since it matches on a variable. So it's probably better to just return `compileCOp opts [] val` directly instead of matching on it.